pkg/project/project.go

```diff
@@ -365,14 +365,22 @@ func (p *Project) loadFromDirectory(
 	p.send(ctx, eventc, LoadEvent{Type: LoadEventStartedWalk})
 
 	err := util.Walk(p.fs, ".", func(path string, info fs.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-
 		ignored := ignoreMatcher.Match(
 			strings.Split(path, string(filepath.Separator)),
 			info.IsDir(),
 		)
+
+		switch err.(type) {
+		case nil:
+		case *os.PathError:
+			if !ignored {
+				p.logger.Warn("path error", zap.String("path", path), zap.Error(err))
+			}
+			ignored = true
+		default:
+			return err
```

Comment:
we should probably return the error to the client to be consistent with https://github.com/stateful/runme/blob/645-runme-attempts-to-access-unreadable-files-or-directories/pkg/project/project.go#L470-L475

Reply:
@sourishkrout If we return the error, it stops the execution. In this case, the approach ignores the permission error to continue walking through the dirs. Let me check if there is a way to return it and continue without halting.

Reply:
i think what you had before was right for the "handled error". i meant, the `default` case which is unhandled errors. when the error is return here will it be returned to the client?

Reply:
@sourishkrout Yes, it will continue working as it did before. It will be handled by the following statement: 

```go
  if err != nil {
		p.send(ctx, eventc, LoadEvent{
			Type: LoadEventError,
			Data: LoadEventErrorData{Err: err},
		})
	}
```	

I'll revert the latest commit.